.npmrc

```diff
@@ -1 +1,2 @@
 @hypertrace:registry=https://hypertrace.jfrog.io/artifactory/api/npm/npm/
+legacy-peer-deps=true
```

Comment:
This change is for npm 7.
Please to get more context https://github.com/angular/angular-cli/issues/19957

Reply:
We shouldn't need this - the legacy peer dep behavior is warning instead of errors, but we should be resolving any peer dependency issues - we want neither warning nor errors.

Reply:
Also, we can stay on npm 6 for now.

```
 ~/trc/traceable-ui/hypertrace-ui   ng-12  ✔  npm install
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
```

Reply:
unless there's a reason to stay on 6, i'd vote to move to 7 as our default - it's been out for a while and is the default one for current node (16, which is an LTS version).

Reply:
We need to collectively move to npm 7 so that the lockfile version problem doesn't come up.

Also, last time the problem came from dependabot. It is running on npm 6 and ends up creating a lock file of version 1. I had to downgrade to 6 due to this.

Reply:
We don't really need to all upgrade or be in sync - the lockfile can bounce back and forth as we see here. Personally, I've been on npm since maybe march? not sure exactly when, but generally keep my versions up to date.

Reply:
dependabot looks like it supports it at least: https://github.blog/changelog/2021-02-08-dependabot-npm-v7-support/

Reply:
And it looks like we've been using it (this repo is using a v2 lockfile): https://raw.githubusercontent.com/hypertrace/hypertrace-ui/main/package-lock.json

Reply:
Oh cool. that works. I think earlier we would end up creating new lock files with newer versions so it was annoying and decided to move back to 6.

Reply:
Removed!!